integration_tests/src/bin/benchmark/main.rs

```diff
@@ -0,0 +1,302 @@
+        GraphicsContext, Options, OutputVideoOptions, PipelineOutputEndCondition,
+        RegisterInputOptions, RegisterOutputOptions,
+    },
+    queue::{self, QueueInputOptions, QueueOptions},
+    Pipeline,
+};
+
+use compositor_pipeline::pipeline::encoder::ffmpeg_h264::Options as H264OutputOptions;
+use compositor_render::{
+    scene::{
+        Component, HorizontalAlign, InputStreamComponent, RGBAColor, TilesComponent, VerticalAlign,
+    },
+    web_renderer::WebRendererInitOptions,
+    Framerate, InputId, OutputId, Resolution,
+};
+use live_compositor::{
+    config::{read_config, LoggerConfig},
+    logger,
+};
+use tracing::warn;
+
+mod args;
+
+use args::{Args, Argument, SingleBenchConfig};
+
+fn main() {
+    let args = Args::parse();
+    let config = read_config();
+    ffmpeg_next::format::network::init();
+    let logger_config = LoggerConfig {
+        level: "compositor_pipeline=error,vk-video=info,benchmark=info".into(),
+        ..config.logger
+    };
+    logger::init_logger(logger_config);
+
+    let ctx = GraphicsContext::new(
+        false,
+        wgpu::Features::SAMPLED_TEXTURE_AND_STORAGE_BUFFER_ARRAY_NON_UNIFORM_INDEXING
+            | wgpu::Features::UNIFORM_BUFFER_AND_STORAGE_TEXTURE_ARRAY_NON_UNIFORM_INDEXING,
+        Default::default(),
+        None,
+    )
+    .unwrap();
+
+    if cfg!(debug_assertions) {
+        warn!("This benchmark is running in debug mode. Make sure to run in release mode for reliable results.");
+    }
+
+    let reports = run_args(ctx, &args);
+    SingleBenchConfig::log_report_header();
+    for report in reports {
+        report.log_as_report();
+    }
+}
+
+fn run_args(ctx: GraphicsContext, args: &Args) -> Vec<SingleBenchConfig> {
+    let arguments = args.arguments();
+    let mut reports = Vec::new();
+
+    // check maximize count
+    let maximize_count = arguments
+        .iter()
+        .filter(|arg| matches!(arg, Argument::Maximize))
+        .count();
+
+    if maximize_count > 1 {
+        panic!("Only one argument can be set to 'maximize'");
+    }
+
+    run_args_iterate(ctx, args, arguments, &mut reports);
+
+    reports
+}
+
+fn run_args_iterate(
+    ctx: GraphicsContext,
+    args: &Args,
+    arguments: Box<[Argument]>,
+    reports: &mut Vec<SingleBenchConfig>,
+) -> bool {
+    for (i, argument) in arguments.iter().enumerate() {
+        if matches!(argument, Argument::IterateExp) {
+            let mut any_succeeded = false;
+            let mut count = 1;
+            loop {
+                let mut arguments = arguments.clone();
+                arguments[i] = Argument::Constant(count);
+
+                if run_args_iterate(ctx.clone(), args, arguments, reports) {
+                    any_succeeded = true;
+                    count *= 2;
+                    continue;
+                } else {
+                    return any_succeeded;
+                }
+            }
+        }
+    }
+
+    run_args_maximize(ctx, args, arguments, reports)
+}
```

Comment:
I'm not sure about this code. On the first sight I thought there might be an infinite recursion. While now I understand what's happening here, the flow here is hard to follow. Also, I tried to run `--decoder-count` with `iterate_exp` and it didn't do anything. I'm not sure if I used it correctly tho